internal/controllers/user_integrations_controller.go

```diff
@@ -220,6 +220,33 @@ func (udc *UserDevicesController) GetIntegrations(c *fiber.Ctx) error {
 	})
 }
 
+// GetIntegrationByTokenID godoc
+// @Description gets an integration using it's tokenID
+// @Tags        integrations
+// @Produce     json
+// @Success     200 {array} ddgrpc.Integration
+// @Security    BearerAuth
+// @Router      /integration/:tokenID [get]
+func (udc *UserDevicesController) GetIntegrationNFTMetadata(c *fiber.Ctx) error {
+	tokenID := c.Params("tokenID")
+
+	uTokenID, err := strconv.ParseUint(tokenID, 10, 64)
+	if err != nil {
+		return fiber.NewError(fiber.StatusInternalServerError, "failed to get integration")
```

Comment:
I think in this case we can say that the token id isn’t a valid number. Probably not an internal error. (If this happens elsewhere in this file then I am sorry.)

Reply:
Was this resolved?

Reply:
I moved the function to the `nftController`. Its been resolved there